h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java

```diff
@@ -103,6 +109,71 @@ public void initMetalearnerParams(Metalearner.Algorithm algo) {
 
+  public static class StackedEnsembleMRScorer extends MRTask<StackedEnsembleMRScorer> {
+    // IN
+    private Frame scoringFrame;
+    private String seKey;
+    private Job job;
+
+    // OUT
+    private Frame levelOneFrame;
+
+    StackedEnsembleMRScorer(Frame scoringFrame, String seKey, Job j) {
+      this.scoringFrame = scoringFrame;
+      this.seKey = seKey;
+      this.job = j;
+    }
+
+    @Override
+    public void map(Key key) {
+      levelOneFrame = new Frame();
+      Model baseModel = (Model) key.get();
+      Frame basePreds = baseModel.score(
+              scoringFrame,
+              "preds_base_" + seKey + scoringFrame._key,
+              job,
+              false
+      );
+      StackedEnsemble.addModelPredictionsToLevelOneFrame(baseModel, basePreds, levelOneFrame);
+      DKV.remove(basePreds._key); //FIXME: is this necessary? It seems to me that `deleteTempFrameAndItsNonSharedVecs` should take care of it
+      Frame.deleteTempFrameAndItsNonSharedVecs(basePreds, levelOneFrame);
+
+      // FIXME: seems fishy to me and there is probably better solution to this problem
+      // Invoking `FrameUtils.cleanUp` seems to be necessary in multinode setting to prevent leaks.
+      // The leaks are occuring from the keys that were created on the slave[1] nodes - more specifically it appears to be
+      // limited to "adapted missing vectors".
+      /*
+       for (Map.Entry<Key, String>  entry: (java.util.Set < Map.Entry < Key, String>>) baseModel._toDelete.entrySet()) {
+         System.out.println(">>>>> Key ='".concat(entry.getKey().toString()).concat("' value = '").concat(entry.getValue()).concat("'"));
+       }
+      */
+      // prints:
+      /*
+      >>>>> Key ='$04ff11000000ffffffff$nfs:/.././smalldata/testng/prostate_test.csv' value = 'adapted missing vectors' << This is leaked without the cleanup
+      >>>>> Key ='$04ff1c000000ffffffff$nfs:/.././smalldata/testng/prostate_test.csv' value = 'categorically adapted vec'
+      >>>>> Key ='$04ff1d000000ffffffff$nfs:/.././smalldata/testng/prostate_test.csv' value = 'categorically adapted vec'
+      >>>>> Key ='$04ff10000000ffffffff$nfs:/.././smalldata/testng/prostate_test.csv' value = 'adapted missing vectors' << This is leaked without the cleanup
+      */
+      // And leaks both of the "adapted missing vectors" on the slave[1] node.
+      //
+      // [1] By slave node I meant a node that wasn't the one that was directly invoked to run the test
+
+      FrameUtils.cleanUp(baseModel._toDelete);
+    }
+
+    @Override
+    public void reduce(StackedEnsembleMRScorer mrt) {
+      levelOneFrame.add(mrt.levelOneFrame);
+    }
+
+    public Frame getLevelOneFrame() {
+      // levelOneFrame can be uninitialized when we don't have any useful base-models, e.g., metalearner is using only intercept
+      levelOneFrame = levelOneFrame == null ? new Frame() : levelOneFrame;
+      levelOneFrame._key = Key.<Frame>make("preds_levelone_" + seKey + scoringFrame._key);
```

Comment:
if you change the key, you probably have to remove the old key from DKV and add this new one: this is a potential key leak.